EditContext/explainer.md

```diff
@@ -59,22 +59,67 @@ Additionally, the EditContext communicates events driven from text input UI to J
   * Composition start and end events.
   * Text formatting requests that indicate where activity relating to text input, e.g. composition, is taking place.
 
+## EditContext Event flow:
+
+This section describes the sequences of events that get fired on the EditContext and focused element when EditContext has focus and IME is active. In this event sequence, the user types in two characters, then commits to the first IME candidate by hitting 'Space'.
+
+|  Event                | EventTarget        |  Related key in sequence
+| -------------         | -----------------  | -------------------
+|  keydown              | focused element    |  Key 1
+|  compositionstart     | active EditContext |  ...
+|  textupdate           | active EditContext |  ...
+|  textformatupdate     | active EditContext |  ...
+|  keyup                | focused element    |  ...
+|  keydown              | focused element    |  Key 2
+|  textupdate           | active EditContext |  ...
+|  textformatupdate     | active EditContext |  ...
+|  keyup                | focused element    |  ...
+|  keydown              | focused element    |  Space
+|  textupdate           | active EditContext |  (committed IME characters available in event.updateText)
+|  keyup                | focused element    |  ...
+|  compositionend       | active EditContext |
+
+Note that the composition events are also not fired on the focused element as the composition is operating on the shared buffer that is represented by the EditContext.
+
 ### EditContext WebIDL
 ```javascript
+
+dictionary EditContextTextRangeInit {

```

Comment:
Let's take Domenic's suggestion and eliminate this type.  The TextUpdateEvent can have the range inlined as updateRangeStart, updateRangeEnd and newSelectionStart, newSelectionEnd.

Similarly TextFormatUpdateEvent will have formatRangeStart, formatRangeEnd